src/librustdoc/html/format.rs

```diff
@@ -327,34 +340,34 @@ impl clean::Constant {
 }
 
 impl clean::PolyTrait {
-    fn print(&self) -> impl fmt::Display + '_ {
+    fn print<'a>(&'a self, cache: &'a Cache) -> impl fmt::Display + 'a {
```

Comment:
```suggestion
    fn print(&self, cache: &Cache) -> impl fmt::Display + '_ {
```

It looks like there are several more of these, and I don't want to overwhelm you with too many comments, so I won't leave any more suggestions :)

Reply:
All have the same answer: it doesn't compile without the explicit lifetime. It's super annoying.